bqskit/compiler/registry.py

```diff
@@ -0,0 +1,99 @@
+"""
+The _workflow_registry enables MachineModel or GateSet specific workflows to be
+registered for used in the `bqskit.compile` method.
+
+The _workflow_registry maps MachineModels a dictionary of Workflows which
+are indexed by optimization level. This object should not be accessed directly
+by the user, but instead through the `register_workflow` function.
+
+Example:
+    model_t = SpecificMachineModel(num_qudits, radixes)
+    workflow = [QuickPartitioner(3), NewFangledOptimization()]
+    register_workflow(model_t, workflow, level)
+    ...
+    new_circuit = compile(circuit, model_t, optimization_level=level)
+"""
```

Comment:
The module docstring will be less visible than the `register_workflow` documentation. You should migrate this helpful information to there.